optimum/exporters/openvino/better_transformer_patch.py

```diff
@@ -0,0 +1,25 @@
+#  Copyright 2023 The HuggingFace Team. All rights reserved.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+import logging as log
+
+
+def patch_model_with_bettertransformer(model):
+    try:
+        model = model.to_bettertransformer()
+    except Exception as e:
+        log.warn(f"Cannot apply model.to_bettertransformer because of the exception:\n{e}")
```

Comment:
Shall we throw an exception here and ask user to try a different configuration (w/o SDPA and stateful) that will work?

Reply:
I'm not sure, because lack of transformations in better transformers not always means that model will be converted w/o SDPA:
* transformers 4.36 bring SDPA support from scratch for some popular models (https://huggingface.co/docs/transformers/perf_infer_gpu_one#flashattention-and-memory-efficient-attention-through-pytorchs-scaleddotproductattention)
* some models with remote code like chatglm, qwen contains SDPA by default and have not support in betterstransformer
* also even w/o SDPA, model can be converted in stateful, just it leads non-effective implementation if standard pattern matching on runtime side can not recognize and fuse SDPA inside it

Reply:
Can we check SDPA availability here and try to call  `to_bettertransformer` then?

Reply:
I added more explained warnings, that stateful may be non-effective in this case.
Regarding SDPA checking, general answer no, we can not track function calls inside pytorch, only after torch.jit.trace or check openvino IR, but this check happens before and bettertransformer usage help us guarantee that it will have it.
I added also some check for known for me model and config attributes that allow to understand that model already has sdpa, but, unfortunately, it is implemented not for all models (e.g. models with remote_code on hf hub, that is not compatible with transformers >= 4.36 introduced api)

Reply:
@AlexKoff88, can we consider this conversation resolved?

Reply:
I think so